src/posit/connect/jobs.py

```diff
@@ -0,0 +1,273 @@
+
+        cluster: Required[Optional[str]]
+        """Location where the job runs, either 'Local' or the cluster name."""
+
+        image: Required[Optional[str]]
+        """Location of the content in clustered environments."""
+
+        run_as: Required[str]
+        """UNIX user responsible for executing the job."""
+
+        # Queue and Scheduling Information
+        queue_name: Required[Optional[str]]
+        """Name of the queue processing the job, relevant for scheduled reports."""
+
+        # Job Metadata
+        tag: Required[JobTag]
+        """A tag categorizing the job type. Options are build_jupyter, build_report, build_site, configure_report, git, packrat_restore, python_restore, render_shiny, run_api, run_app, run_bokeh_app, run_dash_app, run_fastapi_app, run_pyshiny_app, run_python_api, run_streamlit, run_tensorflow, run_voila_app, testing, unknown, val_py_ext_pkg, val_r_ext_pkg, and val_r_install."""
+
+    def __init__(self, ctx, parent: Active, **kwargs: Unpack[_Job]):
+        super().__init__(ctx, parent, **kwargs)
+        self._parent = parent
+
+    @property
+    def _endpoint(self) -> str:
+        return self._ctx.url + f"v1/content/{self._parent['guid']}/jobs/{self['key']}"
+
+    def destroy(self) -> None:
+        """Destroy the job.
+
+        Submit a request to kill the job.
+
+        Warnings
+        --------
+        This operation is irreversible.
+
+        Note
+        ----
+        This action requires administrator, owner, or collaborator privileges.
+        """
+        self._ctx.session.delete(self._endpoint)
+
+
+class Jobs(ActiveFinderMethods[Job]):
+    """A collection of jobs."""
+
+    _uid = "key"
+
+    def __init__(self, cls, ctx, parent: Active):
+        super().__init__(cls, ctx, parent)
+        self._parent = parent
+
+    @property
+    def _endpoint(self) -> str:
+        return self._ctx.url + f"v1/content/{self._parent['guid']}/jobs"
+
+    class _FindByRequest(TypedDict, total=False):
+        # Identifiers
+        id: NotRequired[str]
+        """A unique identifier for the job."""
+
+        ppid: NotRequired[Optional[str]]
+        """Identifier of the parent process."""
+
+        pid: NotRequired[str]
+        """Identifier of the process running the job."""
+
+        key: NotRequired[str]
+        """A unique key to identify this job."""
+
+        remote_id: NotRequired[Optional[str]]
+        """Identifier for off-host execution configurations."""
+
+        app_id: NotRequired[str]
+        """Identifier of the parent content associated with the job."""
+
+        variant_id: NotRequired[str]
+        """Identifier of the variant responsible for the job."""
+
+        bundle_id: NotRequired[str]
+        """Identifier of the content bundle linked to the job."""
+
+        # Timestamps
+        start_time: NotRequired[str]
+        """RFC3339 timestamp indicating when the job started."""
+
+        end_time: NotRequired[Optional[str]]
```

Comment:
What's the difference between `NotRequired` and `Optional`?

Reply:
`Optional[FOO]` represents `FOO | None`. 

I'm in the camp of function parameters can be defined using `Optional` (as it hints that the parameter is _optional_), but everything else should be defined as `FOO | None`.

In this case, `NotRequired` is a special "this key does not have to exist within the dictionary". So `end_time` does not have to exist within a `_FindByRequest` object. (But if the key does exist, it can be a `str` or `None`)

Reply:
`NotRequired` indicates that the argument may be supplied or not supplied. If it is not supplied, no default value is set.

`Optional`, on the other hand, means that the value can be either None or the provided type.

For this example, `NotRequired[Optional[str]]` means that you have three options as the caller.
1. `find_by(end_time="2023-05-25T00:10:01-02:00")` - Find any match with this exact timestamp.
2. `find_by(end_time=None)` - Find any match where the end_time is None (`null` in JSON).
3. `find_by()` - Find any match (using other arguments)